openstack/neutron/templates/secret-neutron-swift-injector.yaml

```diff
@@ -8,4 +8,4 @@ metadata:
 type: Opaque
 data:
   OS_USERNAME: {{ $.Values.global.neutron_service_user | default "neutron" | replace "$" "$$" | b64enc }}
-  OS_PASSWORD: {{ $.Values.global.neutron_service_password | default "" | replace "$" "$$" | b64enc }}
+  OS_PASSWORD: {{ $.Values.global.neutron_service_password | default "" | replace "$" "$$" | include "resolve_secret" | b64enc }}
```

Comment:
I think, that won't work (as intended).
` replace "$" "$$" ` should escape the `$` in the password, now it escapes a `$` in the secrets injector url.
AFAIK, there is no way to escape the password anymore, so you might as well replace it with
`{{ $.Values.global.neutron_service_password }}`.
Our passwords should not contain any `$` anymore.

Reply:
"After" would mean a go-template string for the secret-injector to evaluate like the [urlquery here](https://github.com/sapcc/helm-charts/blob/9c42bd414f0b3524514cd5b0317d02daf5527038/openstack/utils/templates/_hosts.tpl#L13)

Reply:
Not quite:
`include "resolve_secret | replace "$" "$$"` is the same as `replace "$" "$$" | include "resolve_secret"`
Both get evaluated on the client-side before there is any secret involved, which may contain a `$`.

That was the motivation behind the macro, which could be changed like this

```
{{- define "_resolve_secret" -}}
    {{- $str := index . 0 -}}
    {{- $add_urlquery := index . 1 -}}
    {{- $escape_dollar := index . 2 -}}
    {{- if (hasPrefix "vault+kvv2" $str) -}}
        {{"{{"}} resolve "{{ $str }}" {{ if $add_urlquery }}| urlquery {{ end }}{{"}}"}}  {{ if $add_urlquery }}| urlquery {{ end }}{{ if $escape_dollar }} | replace "$" "$$" {{ end }} {{"}}"}}
    {{- else -}}
        {{ $str }}
{{- end -}}
{{- end -}}
```

But now that I read the macro, the else case doesn't do any escaping, which is maybe a bug.